### PR TITLE
New repo workflow

### PR DESCRIPTION
**Pull request by cgundy**
Actually merge code into master.

"The bug" was that I previously only merged my branch of code into another branch, not master. 🤦‍♀️

### CODE REVIEW

File: .github/actions/get_repo_name/action.yml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - name: Get repo name
+        id: get-repo-name
+        run: |
+          NEW_REPO=$(git diff origin/master...HEAD open-repositories.txt | grep "^+[a-zA-Z]" | sed 's/+//' || echo "")
+          if [ "$NEW_REPO" = "" ]; then
```
Comment (EnzoPlayer0ne) [inline .github/actions/get_repo_name/action.yml:23]: You should try `-n` instead - [bash reference](https://www.gnu.org/software/bash/manual/html_node/Bash-Conditional-Expressions.html#Bash-Conditional-Expressions)

File: .github/actions/get_repo_name/action.yml [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - name: Get repo name
+        id: get-repo-name
+        run: |
+          NEW_REPO=$(git diff origin/master...HEAD open-repositories.txt | grep "^+[a-zA-Z]" | sed 's/+//' || echo "")
```
[the remark this answers is not in the document]
Comment (cgundy) [reply .github/actions/get_repo_name/action.yml:22]: Oh yes, I like that a lot better 👍

File: .github/actions/get_repo_name/action.yml [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+            echo "No repository added."
+            exit 0
+          fi
+          REPO_COUNT=$(echo "$NEW_REPO" | wc -l)
+          if (( REPO_COUNT > 1 )); then
+            echo "Only one repository can be added at a time. Please create separate PRs."
+            exit 1
+          elif (( REPO_COUNT = 1 )); then
+            echo "new_repo=$NEW_REPO" >> $GITHUB_OUTPUT
+            echo "new_repo_added=true" >> $GITHUB_OUTPUT
```
[the remark this answers is not in the document]
Comment (cgundy) [reply .github/actions/get_repo_name/action.yml:33]: That's a good point. Let me look into this a little more. I forget if there were certain limitations set by github actions.

**Comment by cgundy** [comment]
Test - all checks are run:

Screenshot 2023-03-07 at 11 39 46

File: .github/actions/get_repo_name/action.yml [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+  steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - name: Get repo name
+        id: get-repo-name
+        run: |
+          NEW_REPO=$(curl --silent https://raw.githubusercontent.com/dfinity/repositories-open-to-contributions/master/open-repositories.txt | grep -vFf- open-repositories.txt)
```
Comment (marko-k0) [inline .github/actions/get_repo_name/action.yml:22]: Maybe add `set -euo pipefail`? Also `--retry` with `curl` ?

**Review by marko-k0 — APPROVED** [review]

**Review by levshaATdfinity — APPROVED** [review]

**Pull request merged**